### PR TITLE
Milestone 3: Setup associations

### PR DESCRIPTION
**Pull request by yetemegn-telaye**
### School-library-system milestone 3 setup-associations

**In this milestone I:**

- [ ] Created new classes i.e rentals, books and classroom 
- [ ] Setup association between classes and objects
- [ ] Run the program using command line

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (BenMukebo) [inline book.rb]: - [x] As it's the `many-to-many` (also has-many-through) relationship between Person and Book using the intermediate class `Rental`. You should add the `Getters and Setters` methods for @rentals. ✍🏻 
- Example: `attr_accessor :title, :author, :rentals`

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
+
```
Comment (BenMukebo) [inline rental.rb]: - [x] The same here, 🤔  A rental belongs to a book and also belongs to a person. Let's make sure to add `Getters and Setters` methids for @book and @person instance variables. ✍🏻 
- Exemple: `attr_accessor :date, :book, :person`

File: person.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require './nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (BenMukebo) [inline person.rb]: - [x] The same here, A person has many rentals so, `Getters and Setters`  for @rentals should be present as well. Please add this 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @yetemegn-telaye,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 
- Methods for adding rentals are present. 👍🏻 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [x] When I click the link to the `issues page` link that you added, I get this 404 `error`: I would recommend fixing the boken issue link in the contributing sections: 

```
## 🤝 Contributing

Contributions, issues, and feature requests are welcome!

Feel free to check the [issues page](../../issues/).
```

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @yetemegn-telaye,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by yetemegn-telaye** [comment]
Hi @Kossi-stack , thank you for approving my pull request but could you please update the status of the project on my microverse dashboard to submit? Because the status is still saying changes required.

**Comment by Kossi-stack** [comment]
Hi @yetemegn-telaye, I have approved the project long time ago. Kindly chat me on slack.

**Pull request merged**